src/webui/webui.cpp

```diff
@@ -145,7 +146,33 @@ void WebUI::configure()
     }
 }
 
+bool WebUI::isEnabled() const
+{
+    return m_isEnabled;
+}
+
 bool WebUI::isErrored() const
 {
     return m_isErrored;
 }
+
+QString WebUI::getUrl() const
```

Comment:
Sorry, but this is something alien here. I didn't mean it was a good idea to just stuff everything into this class. This is clearly not a natural property of it, but it only makes sense for the place where you use it.
I meant only basic (and unambiguous) properties, such as `WebUI::hostAddress`, `WebUI::port`, `WebUI::useHttps`.

Reply:
I agree, but there is a catch. `m_httpServer` may be null.
It can be checked in every property, but then we can get garbage `://:` string in output.

Reply:
> I agree, but there is a catch. `m_httpServer` may be null. It can be checked in every property, but then we can get garbage `://:` string in output.

Why do you want/propose to get these properties through the `m_httpServer` at all? Why not just provide them only based on the WebUI settings, i.e. so that they would always be valid (non-empty)?

Reply:
Because:
1. Requires extra fields, effectively duplicated from the server.
2. They still may be empty, so there is no benefit here.
3. Less reliable because you need manually keep em up in sync with the server.

Reply:
I'm afraid this discussion may come to a dead end. Let's go back to the basics. All I had in mind initially was to encapsulate the logic of processing the corresponding settings in the form of class properties. This logic originally assumed that any empty/unset properties get some predefined values. So you shouldn't have to worry about it from the caller side.

Otherwise could you just leave it as it was before you wrote https://github.com/qbittorrent/qBittorrent/pull/19696#discussion_r1355987286?

Reply:
I'd say that approach is unreliable, because logic can easily get out of sync with the WebUI class. Getting the real working values is better/cleaner imo.

Also maybe you misunderstood my statement a bit.

> but then we can get garbage `://:` string in output.

By that I meant it is _technically possibe_, but by current logic we should never end up in this situation.

So

> meant only basic (and unambiguous) properties, such as `WebUI::hostAddress`, `WebUI::port`, `WebUI::useHttps`.

Looks good in general.

Reply:
> Otherwise could you just leave it as it was before you wrote [#19696 (comment)](https://github.com/qbittorrent/qBittorrent/pull/19696#discussion_r1355987286)?

Let me try it another way first. Then if it will not play out, I will roll it back to initial state.